### PR TITLE
[main] Resolve grub2 CVE-2021-3981

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Resolve grub2 CVE-2021-3981

###### Change Log  

- Fix patch CVE-2021-3981

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-3981

###### Test Methodology

- Local Build

### CODE REVIEW

File: SPECS/grub2/grub2.spec [hunk 1]
```diff
@@ -46,6 +46,8 @@ Patch0156:      0156-efilinux-Fix-integer-overflows-in-grub_cmd_initrd.patch
 Patch0157:      0157-linuxefi-fail-kernel-validation-without-shim-protoco.patch
 # Fix to prevent user from overwriting signed grub binary using grub2-install
 Patch0166:      0166-grub-install-disable-support-for-EFI-platforms.patch
+# CVE-2021-3981
+Patch0167:      0167-restore-umask-for-grub-config.patch 
 # Add nopatches for tooling
 Patch1000:      CVE-2021-3418.nopatch
 Patch1001:      CVE-2020-14372.nopatch
```
Comment (oliviacrain) [inline SPECS/grub2/grub2.spec:53]: While you're here- can we remove the `.nopatch` files from this spec, to be consistent with how we handle all new `.nopatch` files?

File: SPECS/grub2/grub2.spec [hunk 2]
```diff
@@ -46,6 +46,8 @@ Patch0156:      0156-efilinux-Fix-integer-overflows-in-grub_cmd_initrd.patch
 Patch0157:      0157-linuxefi-fail-kernel-validation-without-shim-protoco.patch
 # Fix to prevent user from overwriting signed grub binary using grub2-install
 Patch0166:      0166-grub-install-disable-support-for-EFI-platforms.patch
+# CVE-2021-3981
+Patch0167:      0167-restore-umask-for-grub-config.patch
```
Comment (oliviacrain) [inline SPECS/grub2/grub2.spec:50]: Asking because I'm unsure-will our CVE tooling detect this fix?

File: SPECS/grub2/grub2.spec (hunk 1 above)
Comment (henryli001) [reply SPECS/grub2/grub2.spec:53]: What's our policy of handling .nopatch files? I'm stilling seeing a number of nopatch files existing in other specs?

File: SPECS/grub2/grub2.spec (hunk 2 above)
Comment (henryli001) [reply SPECS/grub2/grub2.spec:50]: Need to double check on this. I'm following what was done for CVE-2020-15705 in this package.

Comment (henryli001) [reply SPECS/grub2/grub2.spec:50]: Confirmed. The CVE number is mentioned in the patch file, so it will be detected by our CVE tooling.

Comment (oliviacrain) [reply SPECS/grub2/grub2.spec:50]: Neat, thanks for checking!

File: SPECS/grub2/grub2.spec (hunk 1 above)
Comment (oliviacrain) [reply SPECS/grub2/grub2.spec:53]: We don't need to mention them in the spec file anymore. Our tooling will pick up on the fact that they're in the same folder as the spec. It's a relatively new thing, I think, so there are still remnants of nopatch files in specs. I won't block on this, it's just a good bit of cleanup to do.

Recent example of a PR where nopatches were added but not added to the spec: #3399

**Review by oliviacrain — APPROVED** [review]

File: SPECS/grub2/grub2.spec (hunk 1 above)
Comment (henryli001) [reply SPECS/grub2/grub2.spec:53]: I see. Will remove them

**Pull request merged**